specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2022-02-01-preview/managedcluster.json

```diff
@@ -0,0 +1,1976 @@
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/clusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "The managed cluster resource updated tags.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ManagedClusterUpdateParameters"
+            },
+            "x-ms-parameter-location": "method"
+          }
+        ],
+        "x-ms-examples": {
+          "Patch a managed cluster": {
+            "$ref": "./examples/ManagedClusterPatchOperation_example.json"
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/ManagedCluster"
+            }
+          },
+          "default": {
+            "description": "The detailed error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorModel"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "ManagedCluster"
+        ],
+        "operationId": "ManagedClusters_Delete",
+        "summary": "Deletes a Service Fabric managed cluster resource.",
+        "description": "Delete a Service Fabric managed cluster resource with the specified name.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/clusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          }
+        ],
+        "x-ms-examples": {
+          "Delete a cluster": {
+            "$ref": "./examples/ManagedClusterDeleteOperation_example.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "responses": {
+          "202": {
+            "description": "The request was accepted and the operation will complete asynchronously."
+          },
+          "204": {
+            "description": "The resource was not found."
+          },
+          "200": {
+            "description": "The operation completed successfully."
+          },
+          "default": {
+            "description": "The detailed error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorModel"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabric/managedClusters/{clusterName}/azresiliencystatus": {
```

Comment:
>azresiliencystatus [](http://example.com/codeflow?start=136&length=18)

Is this a proxy resource? I don't see at the minimum: type, id and name. Also, you are modeling a single get you also need a collection get. This API seems to be a singleton, see here:

[https://github.com/Azure/azure-resource-manager-rpc/blob/f1fd954cebb3958a2a2ccb62b949bf60491866e0/v1.0/common-api-contracts.md#singleton-resources](https://github.com/Azure/azure-resource-manager-rpc/blob/f1fd954cebb3958a2a2ccb62b949bf60491866e0/v1.0/common-api-contracts.md#singleton-resources)

Reply:
@jorgecotillo , it would be great if you could let us know more about what the issue is. As Anant explained, this is a GET Api on the cluster itself, to get the az resiliency  status of the cluster.

Reply:
Will this resource type be added to the RP manifest as ProxyOnly? Then please make it derived from ProxyResource definition in [the common-types](https://github.com/Azure/azure-rest-api-specs/blob/927500222855f07383d48b38596c2ca9835da71b/specification/common-types/resource-management/v3/types.json#L91) and follow the single pattern in RPC (the link provided by Jorge above).

Reply:
This resource is indeed a real resource: 
/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabric/managedClusters/{clusterName}, with type managedClusters and name {clusterName}.

and the property: "azresiliencystatus" returns a status of this resource only. 
Not clear on the proxy part.

Reply:
@j5lim no, we are not planning to add it added to the rp manifest as a proxy resource. it will be similar to the path with have for skus in node types: [/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabric/managedClusters/{clusterName}/nodeTypes/{nodeTypeName}/skus](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/stable/2022-01-01/nodetype.json#L263) example: [NodeTypeSkusListOperation_example.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/stable/2022-01-01/examples/NodeTypeSkusListOperation_example.json)

Reply:
Also, the property: "azresiliencystatus" is a property of each cluster (specifying whether a cluster is AZ redundanat or not) and having a collection get does not really makes sense in this case.  That is why we dont have any collection GET APIs for azresiliencystatus

Reply:
Okay, if this is to return some metadata/status not manageable resource, it should be a POST action. Then you can return whatever you need and don't need to support a collection GET. Please refer to the RPC. https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests

Reply:
@j5lim thanks for the review and comments. We've changed the call to post, please take a look.